src/Containers/NutritionFact/NutritionFact.tsx

```diff
@@ -0,0 +1,307 @@
+import React,{useCallback,useMemo,useRef,forwardRef,useEffect,useState} from 'react'
+import styled from 'styled-components'
+import { Mixins } from '../../Utils'
+import {
+    Main,
+    MainInterface,
+} from '../../Utils/BaseStyles';
+import Theme from '../../Themes/ThemeTemplate'
+import {Input} from '../../Inputs/Input/Input'
+
+const EXTRA_PIXEL=1
+const ENTRY_INPUT_WIDTH=20
+const HEADER_ENTRY_INPUT_WIDTH=40
+
+interface IAdditionalProps{
+    /* if the heading entry must to be taken as the reference entry for taking width from */
+    ref?:boolean;
+    /* if the content of the heading entry must be delayed after setting max width of the main container */
+    delay?:boolean;
+}
+
+interface IEntries{
+    /* type of entry, heading or entry */
+    type:string;
+    /* data per line */
+    entry:IEntryProps|IHeadingEntryProps&IAdditionalProps;
+}
+
+export interface INutritionFactProps{
+    entries:IEntries[];
+    editMode?:boolean;
+}
+
+export const NutritionFact:React.FC<INutritionFactProps>=({entries,editMode}):React.ReactElement=>
+{
+    const mainContainerRef=useRef<HTMLDivElement>(null)
+
+    const containerRef=useRef<HTMLDivElement>(null)
```

Comment:
What does this contain versus the main container?

Reply:
this is a ref to the entry of type 'heading' from which we will take its width to set it as max-width for the rootContainer (mainContainer here)